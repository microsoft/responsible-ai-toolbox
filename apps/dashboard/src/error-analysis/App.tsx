// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { ITheme } from "@fluentui/react";
import {
  IExplanationDashboardData,
  ISerializedExplanationData,
  Metrics
} from "@responsible-ai/core-ui";
import {
  ErrorAnalysisDashboard,
  IErrorAnalysisDashboardProps,
  HelpMessageDict
} from "@responsible-ai/error-analysis";
import { Language } from "@responsible-ai/localization";
import React from "react";

import {
  createJsonImportancesGenerator,
  createPredictionsRequestGenerator,
  generateJsonMatrix,
  generateJsonTreeAdultCensusIncome,
  generateJsonTreeBreastCancer,
  generateJsonTreeBreastCancerPrecision,
  generateJsonTreeBreastCancerRecall,
  generateJsonTreeBoston,
  getJsonMatrix,
  getJsonTreeAdultCensusIncome,
  getJsonTreeBreastCancer,
  getJsonTreeBreastCancerPrecision,
  getJsonTreeBreastCancerRecall,
  getJsonTreeBoston,
  DatasetName
} from "./utils";

interface IAppProps {
  dataset: IExplanationDashboardData | ISerializedExplanationData;
  datasetName: string;
  theme: ITheme;
  language: Language;
  version: 1 | 2 | 3;
  classDimension?: 1 | 2 | 3;
}

export class App extends React.Component<IAppProps> {
  private messages: HelpMessageDict = {
    LocalExpAndTestReq: [{ displayText: "LocalExpAndTestReq", format: "text" }],
    LocalOrGlobalAndTestReq: [
      { displayText: "LocalOrGlobalAndTestReq", format: "text" }
    ],
    PredictorReq: [{ displayText: "PredictorReq", format: "text" }],
    TestReq: [{ displayText: "TestReq", format: "text" }]
  };

  public render(): React.ReactNode {
    let dashboardProp: IErrorAnalysisDashboardProps;
    if (this.props.classDimension === 1) {
      return this.get1D();
    }
    if ("categoricalMap" in this.props.dataset) {
      if (this.props.version === 1) {
        return (
          <ErrorAnalysisDashboard
            modelInformation={{ modelClass: "blackbox" }}
            dataSummary={{
              categoricalMap: this.props.dataset.categoricalMap,
              classNames: this.props.dataset.classNames,
              featureNames: this.props.dataset.featureNames
            }}
            testData={this.props.dataset.trainingData}
            predictedY={this.props.dataset.predictedY as any}
            probabilityY={this.props.dataset.probabilityY}
            trueY={this.props.dataset.trueY as any}
            precomputedExplanations={{
              localFeatureImportance: this.props.dataset.localExplanations
            }}
            requestPredictions={
              !this.props.classDimension
                ? undefined
                : createPredictionsRequestGenerator(this.props.classDimension)
            }
            requestDebugML={generateJsonTreeAdultCensusIncome}
            requestMatrix={generateJsonMatrix(DatasetName.AdultCensusIncome)}
            requestImportances={createJsonImportancesGenerator(
              this.props.dataset.featureNames,
              DatasetName.AdultCensusIncome
            )}
            localUrl={""}
            locale={undefined}
            features={this.props.dataset.featureNames}
            errorAnalysisData={{
              maxDepth: 3,
              metric: Metrics.ErrorRate,
              minChildSamples: 21,
              numLeaves: 31
            }}
          />
        );
      }
      const staticTree = getJsonTreeAdultCensusIncome(
        this.props.dataset.featureNames
      );
      const staticMatrix = getJsonMatrix();
      return (
        <ErrorAnalysisDashboard
          modelInformation={{ modelClass: "blackbox" }}
          dataSummary={{
            categoricalMap: this.props.dataset.categoricalMap,
            classNames: this.props.dataset.classNames,
            featureNames: this.props.dataset.featureNames
          }}
          testData={this.props.dataset.trainingData}
          predictedY={this.props.dataset.predictedY as any}
          probabilityY={this.props.dataset.probabilityY}
          trueY={this.props.dataset.trueY as any}
          precomputedExplanations={{
            localFeatureImportance: this.props.dataset.localExplanations
          }}
          localUrl={""}
          locale={undefined}
          features={this.props.dataset.featureNames}
          errorAnalysisData={{
            matrix: staticMatrix.data,
            matrix_features: staticMatrix.features,
            maxDepth: 3,
            metric: Metrics.ErrorRate,
            minChildSamples: 21,
            numLeaves: 31,
            tree: staticTree.data,
            tree_features: staticTree.features
          }}
        />
      );
    }
    if (this.props.version === 1) {
      const dataset = this.props.dataset as IExplanationDashboardData;
      let datasetType: DatasetName = DatasetName.BreastCancer;
      let requestFunction = generateJsonTreeBreastCancer;
      if (this.props.datasetName === "breastCancerRecallData") {
        datasetType = DatasetName.BreastCancerRecall;
        requestFunction = generateJsonTreeBreastCancerRecall;
      } else if (this.props.datasetName === "breastCancerPrecisionData") {
        datasetType = DatasetName.BreastCancerPrecision;
        requestFunction = generateJsonTreeBreastCancerPrecision;
      }
      dashboardProp = {
        ...dataset,
        errorAnalysisData: {
          maxDepth: 3,
          metric: Metrics.ErrorRate,
          minChildSamples: 21,
          numLeaves: 31
        },
        explanationMethod: "mimic",
        features: dataset.dataSummary.featureNames
          ? dataset.dataSummary.featureNames
          : [],
        locale: this.props.language,
        localUrl: "https://www.bing.com/",
        requestDebugML: requestFunction,
        requestImportances: createJsonImportancesGenerator(
          "dataSummary" in this.props.dataset &&
            this.props.dataset.dataSummary.featureNames
            ? this.props.dataset.dataSummary.featureNames
            : [],
          datasetType
        ),
        requestMatrix: generateJsonMatrix(datasetType),
        requestPredictions: !this.props.classDimension
          ? undefined
          : createPredictionsRequestGenerator(this.props.classDimension),
        stringParams: { contextualHelp: this.messages },
        theme: this.props.theme
      };
    } else {
      let requestFunction = getJsonTreeBreastCancer;
      if (this.props.datasetName === "breastCancerRecallData") {
        requestFunction = getJsonTreeBreastCancerRecall;
      } else if (this.props.datasetName === "breastCancerPrecisionData") {
        requestFunction = getJsonTreeBreastCancerPrecision;
      }
      const dataset = this.props.dataset as IExplanationDashboardData;
      const featureNames = dataset.dataSummary.featureNames
        ? dataset.dataSummary.featureNames
        : [];
      const staticTree = requestFunction(featureNames);
      const staticMatrix = getJsonMatrix();
      dashboardProp = {
        ...(this.props.dataset as IExplanationDashboardData),
        errorAnalysisData: {
          matrix: staticMatrix.data,
          matrix_features: staticMatrix.features,
          maxDepth: 3,
          metric: Metrics.ErrorRate,
          minChildSamples: 21,
          numLeaves: 31,
          tree: staticTree.data,
          tree_features: staticTree.features
        },
        explanationMethod: "mimic",
        features: featureNames,
        locale: this.props.language,
        localUrl: "https://www.bing.com/",
        stringParams: { contextualHelp: this.messages },
        theme: this.props.theme
      };
    }
    return <ErrorAnalysisDashboard {...dashboardProp} />;
  }

  private get1D = (): React.ReactNode => {
    let dashboardProp: IErrorAnalysisDashboardProps;
    const dataset = this.props.dataset as IExplanationDashboardData;
    const featureNames = dataset.dataSummary.featureNames || [];
    if (this.props.version === 1) {
      dashboardProp = {
        ...dataset,
        errorAnalysisData: {
          maxDepth: 3,
          metric: Metrics.MeanSquaredError,
          minChildSamples: 21,
          numLeaves: 31
        },
        explanationMethod: "mimic",
        features: featureNames,
        locale: this.props.language,
        localUrl: "https://www.bing.com/",
        requestDebugML: generateJsonTreeBoston,
        requestImportances: createJsonImportancesGenerator(
          "dataSummary" in this.props.dataset &&
            this.props.dataset.dataSummary.featureNames
            ? this.props.dataset.dataSummary.featureNames
            : [],
          DatasetName.Boston
        ),
        requestMatrix: generateJsonMatrix(DatasetName.Boston),
        requestPredictions: !this.props.classDimension
          ? undefined
          : createPredictionsRequestGenerator(this.props.classDimension),
        stringParams: { contextualHelp: this.messages },
        theme: this.props.theme
      };
    } else {
      const staticTree = getJsonTreeBoston(featureNames);
      const staticMatrix = getJsonMatrix();
      dashboardProp = {
        ...dataset,
        errorAnalysisData: {
          matrix: staticMatrix.data,
          matrix_features: staticMatrix.features,
          maxDepth: 3,
          metric: Metrics.MeanSquaredError,
          minChildSamples: 21,
          numLeaves: 31,
          tree: staticTree.data,
          tree_features: staticTree.features
        },
        explanationMethod: "mimic",
        features: featureNames,
        locale: this.props.language,
        localUrl: "https://www.bing.com/",
        stringParams: { contextualHelp: this.messages },
        theme: this.props.theme
      };
    }
    return <ErrorAnalysisDashboard {...dashboardProp} />;
  };
}
