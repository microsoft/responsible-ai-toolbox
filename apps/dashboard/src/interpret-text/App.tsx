// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { ITheme } from "@fluentui/react";
import { ITextExplanationDashboardData } from "@responsible-ai/core-ui";
import {
  TextExplanationDashboard,
  ITextExplanationDashboardProps
} from "@responsible-ai/interpret-text";
import { Language } from "@responsible-ai/localization";
import React from "react";

interface IAppProps {
  dataset: ITextExplanationDashboardData;
  theme: ITheme;
  language: Language;
  version: 1;
}

export class App extends React.Component<IAppProps> {
  public render(): React.ReactNode {
    const dashboardProp: ITextExplanationDashboardProps = {
      dataSummary: this.props.dataset
    };
    return <TextExplanationDashboard {...dashboardProp} />;
  }
}
