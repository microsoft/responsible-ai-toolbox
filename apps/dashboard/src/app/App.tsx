// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { generateRoute } from "@responsible-ai/core-ui";
import { Language } from "@responsible-ai/localization";
import _ from "lodash";
import { ITheme } from "office-ui-fabric-react";
import React from "react";
import { Redirect, generatePath } from "react-router-dom";

import { App as ErrorAnalysis } from "../error-analysis/App";
import { App as Fairness } from "../fairness/App";
import { App as Interpret } from "../interpret/App";
import { App as ModelAssessment } from "../model-assessment/App";

import { AppHeader } from "./AppHeader";
import { applications, IApplications, applicationKeys } from "./applications";
import { IAppSetting, routeKey } from "./IAppSetting";
import { themes } from "./themes";

interface IAppState extends Required<IAppSetting> {
  application: keyof IApplications;
  iteration: number;
}

export class App extends React.Component<IAppSetting, IAppState> {
  public static route = generateRoute(routeKey);
  public constructor(props: IAppSetting) {
    super(props);
    this.state = this.getState({ ...this.props, iteration: 0 });
  }
  public componentDidUpdate(prevProps: IAppSetting): void {
    if (!_.isEqual(prevProps, this.props)) {
      this.setState(
        this.getState({ ...this.props, iteration: this.state.iteration })
      );
    }
  }
  public render(): React.ReactNode {
    const theme: ITheme = themes[this.state.theme];
    return (
      <>
        <AppHeader onSettingChanged={this.onSettingChanged} {...this.state} />
        <div
          style={{
            backgroundColor: theme.semanticColors.bodyBackground,
            borderColor: "gray",
            borderStyle: "solid",
            borderWidth: 10,
            height: "calc(100% - 70px)",
            minHeight: "500px",
            width: "calc(100%-20px)"
          }}
          key={this.state.iteration}
        >
          {this.state.application === "interpret" && (
            <Interpret
              dataset={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].data
              }
              classDimension={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].classDimension
              }
              theme={theme}
              language={Language[this.state.language]}
              version={
                applications[this.state.application].versions[
                  this.state.version
                ]
              }
            />
          )}
          {this.state.application === "fairness" && (
            <Fairness
              dataset={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].data
              }
              theme={themes[this.state.theme]}
              language={Language[this.state.language]}
              version={
                applications[this.state.application].versions[
                  this.state.version
                ]
              }
            />
          )}
          {this.state.application === "errorAnalysis" && (
            <ErrorAnalysis
              dataset={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].data
              }
              classDimension={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].classDimension
              }
              theme={theme}
              language={Language[this.state.language]}
              version={
                applications[this.state.application].versions[
                  this.state.version
                ]
              }
            />
          )}
          {this.state.application === "modelAssessment" && (
            <ModelAssessment
              dataset={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].data
              }
              classDimension={
                applications[this.state.application].datasets[
                  this.state.dataset
                ].classDimension
              }
              theme={theme}
              language={Language[this.state.language]}
              version={
                applications[this.state.application].versions[
                  this.state.version
                ]
              }
            />
          )}
        </div>
        <Redirect to={generatePath(App.route, this.state)} push={true} />
      </>
    );
  }
  private onSettingChanged = <T extends keyof IAppSetting>(
    field: T,
    value: IAppSetting[T]
  ): void => {
    this.setState((prev: IAppState) => {
      const newState = { ...prev, [field]: value };
      return this.getState(newState);
    });
  };

  private getState(props: IAppSetting & { iteration: number }): IAppState {
    const idx = applicationKeys.indexOf(
      props.application as keyof IApplications
    );
    const application: keyof IApplications =
      idx < 0 ? "interpret" : applicationKeys[idx];
    return {
      application,
      dataset:
        !props.dataset || !applications[application].datasets[props.dataset]
          ? Object.keys(applications[application].datasets)[0]
          : props.dataset,
      iteration: props.iteration + 1,
      language:
        !props.language || !Language[props.language]
          ? Language.En
          : props.language,
      theme:
        !props.theme || !themes[props.theme]
          ? Object.keys(themes)[0]
          : props.theme,
      version:
        !props.version || !applications[application].versions[props.version]
          ? Object.keys(applications[application].versions)[0]
          : props.version
    };
  }
}
