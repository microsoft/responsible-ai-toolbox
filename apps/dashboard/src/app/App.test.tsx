// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { shallow } from "enzyme";
import React from "react";

import { App } from "./App";

describe("FairnessWizardV1", () => {
  it("should render successfully", () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
