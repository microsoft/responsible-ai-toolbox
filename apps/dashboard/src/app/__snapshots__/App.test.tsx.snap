// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FairnessWizardV1 should render successfully 1`] = `
<Fragment>
  <AppHeader
    application="interpret"
    dataset="automlMimicAdult"
    iteration={1}
    language="Cs"
    onSettingChanged={[Function]}
    theme="1 - light"
    version="Version-1"
  />
  <div
    key="1"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "gray",
        "borderStyle": "solid",
        "borderWidth": 10,
        "height": "calc(100% - 70px)",
        "minHeight": "500px",
        "width": "calc(100%-20px)",
      }
    }
  >
    <App
      dataset={
        Object {
          "dataSummary": Object {
            "classNames": Array [
              " <=50K",
              " >50K",
            ],
            "featureNames": Array [
              "Column1",
              "Column2",
              "Column3",
              "Column4",
              "Column5",
              "Column6",
              "Column7",
              "Column8",
              "Column9",
              "Column10",
              "Column11",
              "Column12",
              "Column13",
              "Column14",
            ],
          },
          "modelInformation": Object {
            "method": "classifier",
            "modelClass": "blackbox",
          },
          "precomputedExplanations": Object {
            "localFeatureImportance": Object {
              "intercept": Array [
                2.017027712531668,
                -2.017027712531668,
              ],
              "scores": Array [
                Array [
                  Array [
                    0.3983844330617047,
                    -0.007446859076005137,
                    -0.011284607644183499,
                    -0.011639590771705607,
                    0.13797249559423214,
                    1.1299295633569393,
                    0.10483989218975776,
                    -0.07662816714907031,
                    -0.00990259974787941,
                    0.19384559313235822,
                    0.1231644403583161,
                    0.029931112742487255,
                    0.15630157046311205,
                    -0.011875052435833713,
                  ],
                  Array [
                    0.9121102147381127,
                    -0.014893369015455753,
                    0.06288009358930821,
                    0.03236953362735648,
                    -0.7303862839495617,
                    1.3777566009250506,
                    -0.2314539971140443,
                    -0.13337987246139965,
                    -0.00350515816823669,
                    0.11738792312770502,
                    0.1388117775371375,
                    0.0349846964039984,
                    0.39774693763619484,
                    -0.008223446249958626,
                  ],
                  Array [
                    -0.6909525158636243,
                    -0.21172909591586708,
                    -0.00018149000874417652,
                    0.024230902003562402,
                    -0.8473793466799153,
                    1.0482247382115149,
                    -0.23459771884396824,
                    -0.0845509300999351,
                    0.11065070705422428,
                    0.22189543376231105,
                    0.17377892670293218,
                    0.03111581380124407,
                    0.09564100246642705,
                    -0.010629925467898777,
                  ],
                  Array [
                    0.036326586279450464,
                    -0.010689275128596845,
                    -0.011500723210033805,
                    0.009429516614012744,
                    -0.8029923787291721,
                    1.4041147883221359,
                    0.24399358737609791,
                    -0.10397518612882117,
                    -0.01158925720009654,
                    0.1805276579046236,
                    0.18477344522408906,
                    0.0351527694385445,
                    0.1650821745932958,
                    -0.01170048391572412,
                  ],
                  Array [
                    0.03111701839051829,
                    0.12687720782761477,
                    -0.008625702891362825,
                    -0.010628941903024491,
                    0.16465828862577583,
                    0.8764196131226375,
                    0.045151988362574516,
                    -0.11176953189415802,
                    -0.007084609045849493,
                    -0.1475309269363674,
                    0.4252504847221428,
                    0.03137305059960977,
                    -0.6207376822287531,
                    -0.010903900381217699,
                  ],
                ],
                Array [
                  Array [
                    -0.3983844330617047,
                    0.007446859076005137,
                    0.011284607644183499,
                    0.011639590771705607,
                    -0.13797249559423214,
                    -1.1299295633569393,
                    -0.10483989218975776,
                    0.07662816714907031,
                    0.00990259974787941,
                    -0.19384559313235822,
                    -0.1231644403583161,
                    -0.029931112742487255,
                    -0.15630157046311205,
                    0.011875052435833713,
                  ],
                  Array [
                    -0.9121102147381127,
                    0.014893369015455753,
                    -0.06288009358930821,
                    -0.03236953362735648,
                    0.7303862839495617,
                    -1.3777566009250506,
                    0.2314539971140443,
                    0.13337987246139965,
                    0.00350515816823669,
                    -0.11738792312770502,
                    -0.1388117775371375,
                    -0.0349846964039984,
                    -0.39774693763619484,
                    0.008223446249958626,
                  ],
                  Array [
                    0.6909525158636243,
                    0.21172909591586708,
                    0.00018149000874417652,
                    -0.024230902003562402,
                    0.8473793466799153,
                    -1.0482247382115149,
                    0.23459771884396824,
                    0.0845509300999351,
                    -0.11065070705422428,
                    -0.22189543376231105,
                    -0.17377892670293218,
                    -0.03111581380124407,
                    -0.09564100246642705,
                    0.010629925467898777,
                  ],
                  Array [
                    -0.036326586279450464,
                    0.010689275128596845,
                    0.011500723210033805,
                    -0.009429516614012744,
                    0.8029923787291721,
                    -1.4041147883221359,
                    -0.24399358737609791,
                    0.10397518612882117,
                    0.01158925720009654,
                    -0.1805276579046236,
                    -0.18477344522408906,
                    -0.0351527694385445,
                    -0.1650821745932958,
                    0.01170048391572412,
                  ],
                  Array [
                    -0.03111701839051829,
                    -0.12687720782761477,
                    0.008625702891362825,
                    0.010628941903024491,
                    -0.16465828862577583,
                    -0.8764196131226375,
                    -0.045151988362574516,
                    0.11176953189415802,
                    0.007084609045849493,
                    0.1475309269363674,
                    -0.4252504847221428,
                    -0.03137305059960977,
                    0.6207376822287531,
                    0.010903900381217699,
                  ],
                ],
              ],
            },
          },
        }
      }
      language="Cs"
      theme={
        Object {
          "disableGlobalClassNames": false,
          "effects": Object {
            "elevation16": "0 6.4px 14.4px 0 rgba(0, 0, 0, 0.132), 0 1.2px 3.6px 0 rgba(0, 0, 0, 0.108)",
            "elevation4": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
            "elevation64": "0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18)",
            "elevation8": "0 3.2px 7.2px 0 rgba(0, 0, 0, 0.132), 0 0.6px 1.8px 0 rgba(0, 0, 0, 0.108)",
            "roundedCorner2": "2px",
            "roundedCorner4": "4px",
            "roundedCorner6": "6px",
          },
          "fonts": Object {
            "large": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "18px",
              "fontWeight": 400,
            },
            "medium": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "14px",
              "fontWeight": 400,
            },
            "mediumPlus": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "16px",
              "fontWeight": 400,
            },
            "mega": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "68px",
              "fontWeight": 600,
            },
            "small": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "12px",
              "fontWeight": 400,
            },
            "smallPlus": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "12px",
              "fontWeight": 400,
            },
            "superLarge": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "42px",
              "fontWeight": 600,
            },
            "tiny": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "10px",
              "fontWeight": 400,
            },
            "xLarge": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "20px",
              "fontWeight": 600,
            },
            "xLargePlus": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "24px",
              "fontWeight": 600,
            },
            "xSmall": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "10px",
              "fontWeight": 400,
            },
            "xxLarge": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "28px",
              "fontWeight": 600,
            },
            "xxLargePlus": Object {
              "MozOsxFontSmoothing": "grayscale",
              "WebkitFontSmoothing": "antialiased",
              "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
              "fontSize": "32px",
              "fontWeight": 600,
            },
          },
          "isInverted": false,
          "palette": Object {
            "accent": "#0078d4",
            "black": "#000000",
            "blackTranslucent40": "rgba(0,0,0,.4)",
            "blue": "#0078d4",
            "blueDark": "#002050",
            "blueLight": "#00bcf2",
            "blueMid": "#00188f",
            "green": "#107c10",
            "greenDark": "#004b1c",
            "greenLight": "#bad80a",
            "magenta": "#b4009e",
            "magentaDark": "#5c005c",
            "magentaLight": "#e3008c",
            "neutralDark": "#201f1e",
            "neutralLight": "#edebe9",
            "neutralLighter": "#f3f2f1",
            "neutralLighterAlt": "#faf9f8",
            "neutralPrimary": "#323130",
            "neutralPrimaryAlt": "#3b3a39",
            "neutralQuaternary": "#d0d0d0",
            "neutralQuaternaryAlt": "#e1dfdd",
            "neutralSecondary": "#605e5c",
            "neutralSecondaryAlt": "#8a8886",
            "neutralTertiary": "#a19f9d",
            "neutralTertiaryAlt": "#c8c6c4",
            "orange": "#d83b01",
            "orangeLight": "#ea4300",
            "orangeLighter": "#ff8c00",
            "purple": "#5c2d91",
            "purpleDark": "#32145a",
            "purpleLight": "#b4a0ff",
            "red": "#e81123",
            "redDark": "#a4262c",
            "teal": "#008272",
            "tealDark": "#004b50",
            "tealLight": "#00b294",
            "themeDark": "#005a9e",
            "themeDarkAlt": "#106ebe",
            "themeDarker": "#004578",
            "themeLight": "#c7e0f4",
            "themeLighter": "#deecf9",
            "themeLighterAlt": "#eff6fc",
            "themePrimary": "#0078d4",
            "themeSecondary": "#2b88d8",
            "themeTertiary": "#71afe5",
            "white": "#ffffff",
            "whiteTranslucent40": "rgba(255,255,255,.4)",
            "yellow": "#ffb900",
            "yellowDark": "#d29200",
            "yellowLight": "#fff100",
          },
          "rtl": undefined,
          "semanticColors": Object {
            "accentButtonBackground": "#0078d4",
            "accentButtonText": "#ffffff",
            "actionLink": "#323130",
            "actionLinkHovered": "#201f1e",
            "blockingBackground": "#FDE7E9",
            "blockingIcon": "#FDE7E9",
            "bodyBackground": "#ffffff",
            "bodyBackgroundChecked": "#edebe9",
            "bodyBackgroundHovered": "#f3f2f1",
            "bodyDivider": "#edebe9",
            "bodyFrameBackground": "#ffffff",
            "bodyFrameDivider": "#edebe9",
            "bodyStandoutBackground": "#faf9f8",
            "bodySubtext": "#605e5c",
            "bodyText": "#323130",
            "bodyTextChecked": "#000000",
            "buttonBackground": "#ffffff",
            "buttonBackgroundChecked": "#c8c6c4",
            "buttonBackgroundCheckedHovered": "#edebe9",
            "buttonBackgroundDisabled": "#f3f2f1",
            "buttonBackgroundHovered": "#f3f2f1",
            "buttonBackgroundPressed": "#edebe9",
            "buttonBorder": "#8a8886",
            "buttonBorderDisabled": "#f3f2f1",
            "buttonText": "#323130",
            "buttonTextChecked": "#201f1e",
            "buttonTextCheckedHovered": "#000000",
            "buttonTextDisabled": "#a19f9d",
            "buttonTextHovered": "#201f1e",
            "buttonTextPressed": "#201f1e",
            "cardShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
            "cardShadowHovered": "0 3.2px 7.2px 0 rgba(0, 0, 0, 0.132), 0 0.6px 1.8px 0 rgba(0, 0, 0, 0.108)",
            "cardStandoutBackground": "#ffffff",
            "defaultStateBackground": "#faf9f8",
            "disabledBackground": "#f3f2f1",
            "disabledBodySubtext": "#c8c6c4",
            "disabledBodyText": "#a19f9d",
            "disabledBorder": "#c8c6c4",
            "disabledSubtext": "#d0d0d0",
            "disabledText": "#a19f9d",
            "errorBackground": "#FDE7E9",
            "errorIcon": "#A80000",
            "errorText": "#a4262c",
            "focusBorder": "#605e5c",
            "infoBackground": "#f3f2f1",
            "infoIcon": "#605e5c",
            "inputBackground": "#ffffff",
            "inputBackgroundChecked": "#0078d4",
            "inputBackgroundCheckedHovered": "#005a9e",
            "inputBorder": "#605e5c",
            "inputBorderHovered": "#323130",
            "inputFocusBorderAlt": "#0078d4",
            "inputForegroundChecked": "#ffffff",
            "inputIcon": "#0078d4",
            "inputIconDisabled": "#a19f9d",
            "inputIconHovered": "#005a9e",
            "inputPlaceholderBackgroundChecked": "#deecf9",
            "inputPlaceholderText": "#605e5c",
            "inputText": "#323130",
            "inputTextHovered": "#201f1e",
            "link": "#0078d4",
            "linkHovered": "#004578",
            "listBackground": "#ffffff",
            "listHeaderBackgroundHovered": "#f3f2f1",
            "listHeaderBackgroundPressed": "#edebe9",
            "listItemBackgroundChecked": "#edebe9",
            "listItemBackgroundCheckedHovered": "#e1dfdd",
            "listItemBackgroundHovered": "#f3f2f1",
            "listText": "#323130",
            "listTextColor": "#323130",
            "menuBackground": "#ffffff",
            "menuDivider": "#c8c6c4",
            "menuHeader": "#0078d4",
            "menuIcon": "#0078d4",
            "menuItemBackgroundChecked": "#edebe9",
            "menuItemBackgroundHovered": "#f3f2f1",
            "menuItemBackgroundPressed": "#edebe9",
            "menuItemText": "#323130",
            "menuItemTextHovered": "#201f1e",
            "messageLink": "#005A9E",
            "messageLinkHovered": "#004578",
            "messageText": "#323130",
            "primaryButtonBackground": "#0078d4",
            "primaryButtonBackgroundDisabled": "#f3f2f1",
            "primaryButtonBackgroundHovered": "#106ebe",
            "primaryButtonBackgroundPressed": "#005a9e",
            "primaryButtonBorder": "transparent",
            "primaryButtonText": "#ffffff",
            "primaryButtonTextDisabled": "#d0d0d0",
            "primaryButtonTextHovered": "#ffffff",
            "primaryButtonTextPressed": "#ffffff",
            "severeWarningBackground": "#FED9CC",
            "severeWarningIcon": "#D83B01",
            "smallInputBorder": "#605e5c",
            "successBackground": "#DFF6DD",
            "successIcon": "#107C10",
            "successText": "#107C10",
            "variantBorder": "#edebe9",
            "variantBorderHovered": "#a19f9d",
            "warningBackground": "#FFF4CE",
            "warningHighlight": "#ffb900",
            "warningIcon": "#797775",
            "warningText": "#323130",
          },
          "spacing": Object {
            "l1": "20px",
            "l2": "32px",
            "m": "16px",
            "s1": "8px",
            "s2": "4px",
          },
        }
      }
      version={1}
    />
  </div>
  <Redirect
    push={true}
    to="/interpret/automlMimicAdult/1%20-%20light/Cs/Version-1"
  />
</Fragment>
`;
